service-base/src/main/java/org/eclipse/hono/service/metric/MicrometerBasedMetrics.java

```diff
@@ -79,6 +84,8 @@
     protected final MeterRegistry registry;
 
     private final Map<String, AtomicLong> authenticatedConnections = new ConcurrentHashMap<>();
+    private final Map<String, AtomicLong> authenticatedConnectionsTime = new ConcurrentHashMap<>();
```

Comment:
why do we need two maps here? The DeviceConnectionTimeTracker already contains the number of connections, doesn't it?

Reply:
`DeviceConnectionTimeTracker` has already number of connections. The map above is to keep the reference to the gauge state object that records the connection time. 

Micrometer documentation says that "It is your responsibility to hold a strong reference to the state object that you are measuring with a Gauge.  Once the object being gauged is de-referenced and is garbage collected, Micrometer will start reporting a NaN or nothing for a gauge". This is the situation I face, if I try to maintain the reference within the `DeviceConnectionTimer`. When the number of connections are zero, null is returned and this leads to garbage collection of the gauge state object reference and resulting in NaN and henceforth the values are not recorded in Prometheus. 

To overcome this, I am maintaining the state object references in a map here in paar with maps used to store the references of the number of connections.

Reply:
The more I think about it the less convinced I am that using a Gauge makes much sense. I think your original approach using a Timer is the right choice as it better reflects the nature of what we're doing here of reporting distinct amounts of (connection) time whereas a Gauge is more like a counter which, in our case, would need to continuously be increased by the latest amount of time. However, this way the value would increase forever until it would eventually exceed the MAX or we would need to reset it periodically. All of that doesn't really feel right so IMHO going back to the Timer is the right thing to do. Sorry for the inconvenience this has caused but I was hoping to get away with a less resource hungry alternative ...

Reply:
Considering the lower memory footprint of _Gauge_, it was worth a try. Now I will revert to the original approach using a _Timer_.

Reply:
I have reverted back to the _Timer_ approach and pushed those changes. Would you mind taking a look again? Meanwhile I will also update the documentation.